service/src/main/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceImpl.kt

```diff
@@ -665,23 +664,44 @@ internal class SøknadsbehandlingServiceImpl(
             grunnlagService.lagreBosituasjongrunnlag(behandlingId = request.behandlingId, listOf(bosituasjon))
             return when (it) {
                 is Søknadsbehandling.Vilkårsvurdert.Avslag -> it.copy(
-                    grunnlagsdata = it.grunnlagsdata.copy(
-                        bosituasjon = listOf(
-                            bosituasjon,
-                        ),
+                    grunnlagsdata = Grunnlagsdata.tryCreate(
+                        bosituasjon = listOf(bosituasjon),
+                        fradragsgrunnlag = it.grunnlagsdata.fradragsgrunnlag,
                     ),
                 )
                 is Søknadsbehandling.Vilkårsvurdert.Innvilget -> it.copy(
-                    grunnlagsdata = it.grunnlagsdata.copy(
+                    grunnlagsdata = Grunnlagsdata.tryCreate(
                         bosituasjon = listOf(bosituasjon),
+                        fradragsgrunnlag = it.grunnlagsdata.fradragsgrunnlag,
                     ),
                 )
                 is Søknadsbehandling.Vilkårsvurdert.Uavklart -> it.copy(
-                    grunnlagsdata = it.grunnlagsdata.copy(
+                    grunnlagsdata = Grunnlagsdata.tryCreate(
                         bosituasjon = listOf(bosituasjon),
+                        fradragsgrunnlag = it.grunnlagsdata.fradragsgrunnlag,
                     ),
                 )
             }.right()
         }
     }
+
+    override fun leggTilFradragGrunnlag(request: LeggTilFradragsgrunnlagRequest): Either<KunneIkkeLeggeTilFradragsgrunnlag, Søknadsbehandling> {
+        val behandling = søknadsbehandlingRepo.hent(request.behandlingId)
+            ?: return KunneIkkeLeggeTilFradragsgrunnlag.FantIkkeBehandling.left()
+
+        val oppdatertBehandling = behandling.leggTilFradragsgrunnlag(request.fradragsrunnlag).getOrHandle {
+            return when (it) {
+                Søknadsbehandling.KunneIkkeLeggeTilFradragsgrunnlag.GrunnlagetMåVæreInneforBehandlingen -> KunneIkkeLeggeTilFradragsgrunnlag.FradragsgrunnlagUtenforPeriode.left()
+                Søknadsbehandling.KunneIkkeLeggeTilFradragsgrunnlag.IkkeLovÅLeggeTilFradragIDenneStatusen -> KunneIkkeLeggeTilFradragsgrunnlag.UgyldigTilstand(
+                    fra = behandling::class,
+                ).left()
+                Søknadsbehandling.KunneIkkeLeggeTilFradragsgrunnlag.PeriodeMåFyllesUt -> KunneIkkeLeggeTilFradragsgrunnlag.PeriodeMangler.left()
```

Comment:
Stemmer for å bruke samme navn på venstre - høyre side. Bare velg det som leser best